eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -588,30 +588,32 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
     }
-    // We check whether this is the shared input or a remote input.
-    val input = addInput.previousTx_opt match {
-      case Some(previousTx) if previousTx.txOut.length <= addInput.previousTxOutput =>
-        return Left(InputOutOfBounds(fundingParams.channelId, addInput.serialId, previousTx.txid, addInput.previousTxOutput))
-      case Some(previousTx) if fundingParams.sharedInput_opt.exists(_.info.outPoint == OutPoint(previousTx, addInput.previousTxOutput.toInt)) =>
-        return Left(InvalidSharedInput(fundingParams.channelId, addInput.serialId))
-      case Some(previousTx) if !Script.isNativeWitnessScript(previousTx.txOut(addInput.previousTxOutput.toInt).publicKeyScript) =>
-        return Left(NonSegwitInput(fundingParams.channelId, addInput.serialId, previousTx.txid, addInput.previousTxOutput))
-      case Some(previousTx) =>
-        Input.Remote(addInput.serialId, OutPoint(previousTx, addInput.previousTxOutput.toInt), previousTx.txOut(addInput.previousTxOutput.toInt), addInput.sequence)
-      case None =>
-        (addInput.sharedInput_opt, fundingParams.sharedInput_opt) match {
-          case (Some(outPoint), Some(sharedInput)) if outPoint == sharedInput.info.outPoint =>
-            Input.Shared(addInput.serialId, outPoint, sharedInput.info.txOut.publicKeyScript, addInput.sequence, purpose.previousLocalBalance, purpose.previousRemoteBalance, purpose.htlcBalance)
-          case _ =>
-            return Left(PreviousTxMissing(fundingParams.channelId, addInput.serialId))
-        }
+    // We check whether this is the shared input or a remote input, and validate input details if it is not the shared input.
+    // The remote input details are usually provided by sending the entire previous transaction.
+    // But when splicing a taproot channel, it is possible to only send the previous txOut (which saves bandwidth and
+    // allows spending transactions that are larger than 65kB) because the signature of the shared taproot input will
+    // commit to *every* txOut that is being spent, which protects against malleability issues.
+    // See https://delvingbitcoin.org/t/malleability-issues-when-creating-shared-transactions-with-segwit-v0/497 for more details.
+    val remoteInputInfo_opt = (addInput.previousTx_opt, addInput.previousTxOut_opt) match {
+      case (Some(previousTx), _) if previousTx.txOut.length <= addInput.previousTxOutput => return Left(InputOutOfBounds(fundingParams.channelId, addInput.serialId, previousTx.txid, addInput.previousTxOutput))
+      case (Some(previousTx), _) => Some(InputInfo(OutPoint(previousTx, addInput.previousTxOutput.toInt), previousTx.txOut(addInput.previousTxOutput.toInt)))
+      case (None, Some(_)) if !fundingParams.sharedInput_opt.exists(_.commitmentFormat.isInstanceOf[TaprootCommitmentFormat]) => return Left(PreviousTxMissing(fundingParams.channelId, addInput.serialId))
+      case (None, Some(inputInfo)) => Some(inputInfo)
+      case (None, None) => None
```

Comment:
nit: does it make sense to allow both `previousTx_opt` and `previousTxOut_opt` to be present ?